FilmFestivalLoader/IDFA/parse_idfa_html.py

```diff
@@ -258,89 +494,248 @@ def add_paragraph(self):
     def set_article(self):
         self.article = '\n\n'.join(self.article_paragraphs)
         self.set_description_from_article(self.film.title)
+        counter.increase('articles')
 
     def add_film_info(self):
-        film_info = planner.FilmInfo(self.film.filmid, self.description, self.article)
-        self.festival_data.filminfos.append(film_info)
+        self.film_info = FilmInfo(self.film.filmid, self.description, self.article)
+        self.festival_data.filminfos.append(self.film_info)
+
+    def update_film_info(self):
+        counter.increase('filminfo update')
+        if self.film_property_by_label:
+            counter.increase('meta dicts')
+            properties = [f'{key}: {value}' for (key, value) in self.film_property_by_label.items()]
+            metadata = '\n'.join(properties)
+            self.film_info.article += f'\n\n{metadata}'
+            counter.increase('filminfo extended')
+
+    def set_screening_date(self, data):
+        parts = data.split()  # '10 november'
+        try:
+            day = int(parts[0])
+            month = int(FilmPageParser.nl_month_by_name[parts[1]])
+        except ValueError as e:
+            counter.increase('improper dates')
+            self.print_debug(f'{e} in {self.film}', 'Proceeding to next page section')
+            return False
+        else:
+            self.start_date = datetime.date(day=day, month=month, year=festival_year)
+        return True
 
     def set_screening_times(self, data):
-        self.start_dt, self.end_dt = get_screening_times(data)
+        try:
+            start_time = datetime.time(int(data[:2]), int(data[3:5]))   # '14.00–15.28'
+            end_time = datetime.time(int(data[6:8]), int(data[9:]))
+        except ValueError as e:
+            counter.increase('improper times')
+            self.print_debug(f'{e} in times of {self.film} screening', 'Proceeding to next page section')
+            return False
+        else:
+            start_date = self.start_date
+            end_date = start_date if end_time > start_time else start_date + datetime.timedelta(days=1)
+            self.start_dt = datetime.datetime.combine(start_date, start_time)
+            self.end_dt = datetime.datetime.combine(end_date, end_time)
+        return True
 
     def get_screen(self, data):
-        screen_name = data.strip()
-        screen = self.festival_data.get_screen(festival_city, screen_name)
+        screen_parse_name = data.strip()
+        city_name, theater_parse_name, screen_abbreviation = self.split_location(screen_parse_name)
+        screen = self.festival_data.get_screen(
+            city_name,
+            screen_parse_name,
+            theater_parse_name=theater_parse_name,
+            screen_abbreviation=screen_abbreviation
+        )
         return screen
```

Comment:
Please make this common code. The exact same code is included in `Imagine_html_parser.py`.

Reply:
Done, also in `parse_imagine_html.py`.